extractuser.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"crypto/rsa"
+
+	"golang.org/x/net/context"
+
+	"net/http"
+
+	"github.com/almighty/almighty-core/token"
+	"github.com/goadesign/goa"
+)
+
+// ConfigureExtractUser is a configuration for middleware,
+// Accepts PublicKey and PrivateKey to create a token manager
+// Using token manager, it is responsible for extracting the token from context if present
+// Update the context with uuid found in token
+func ConfigureExtractUser(publicKey *rsa.PublicKey, privateKey *rsa.PrivateKey) (ExtractUser goa.Middleware) {
+	manager := token.NewManager(publicKey, privateKey)
+	return func(h goa.Handler) goa.Handler {
+		return func(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+			// for now ignore the error becasue still test for logged in user is not done.
+			uuid, err := manager.Locate(ctx)
+			var value string
+			if err != nil {
+				value = "guest user"
+			}
+			value = uuid.String()
+			ctxWithUser := context.WithValue(ctx, "uuid", value)
```

Comment:
Thanks @aslakknutsen for input, now added 2 new functions for that.
And `key` renamed to `CurrentUser`


Done in [7e64ea](https://github.com/almighty/almighty-core/pull/387/commits/7e64ea82268e1db1d02cf4f67bca25c2563f1520)